app/controllers/user/masquerades_controller.rb

```diff
@@ -0,0 +1,14 @@
+class User::MasqueradesController < Devise::MasqueradesController
+  before_action :load_and_authorize_user, only: [:show]
```

Comment:
There are only 2 actions in the parent controller, `#show` and `#back`, `#back` is for stop masquerading, think we don't need to authorise that ?  There is no user_id in `#back` also.